spectral_cube/stokes_spectral_cube.py

```diff
@@ -34,8 +50,15 @@ def __init__(self, stokes_data, mask=None, meta=None, fill_value=None):
 
         reference = tuple(stokes_data.keys())[0]
 
-        for component in stokes_data:
+        # Validate and map Stokes components using StokesCoord, with custom mapping
+        # Only catch ValueError from StokesCoord, not all exceptions
+        try:
+            with custom_stokes_symbol_mapping(self._custom_stokes_map):
+                stokes_coord = StokesCoord(list(stokes_data.keys()))
+        except ValueError as e:
+            raise ValueError(f"Invalid Stokes components: {e}")
```

Comment:
Is the new value error more clear than the original?  If so, fine, but I would kind of expect `StokesCoord` to raise its own entirely valid & useful error message here, such that the try/except is redundant.  But maybe this is actually helpful.